tests/test_ecs.py

```diff
@@ -100,6 +100,44 @@ def test_ecs_service_metrics(self):
         self.assertEqual(len(resources), 1)
         self.assertTrue("c7n.metrics" in resources[0])
 
+    def test_ecs_service_update(self):
+        session_factory = self.replay_flight_data("test_ecs_service_update")
+        client = session_factory().client("ecs")
+        svc = client.describe_services(
```

Comment:
rather than making a separate call out here to describe the service state, you can just filter in services that have assign public ip enabled

Reply:
I'm slightly confused. Is the issue the code is being inefficient in making more APIs calls than it has to? Should I reach across methods to save API calls (ie pass in an already used string into replay_flight_data())? The pattern I've been using is:
1) API call to confirm prior value
2) Run Policy
3) API call to confirm value changed. 
Are you saying I can just skip step 1)? Or replace 1) with a custodian run (instead of a direct client.describe_services call)?

Reply:
rather than doing step 1, just write the policy in such a way that you dont need an additional api call to confirm the prior value. e.g.

```yaml
policies:
  - name: public-ip-service
    resource: ecs-service
    filters:
      - networkConfiguration.awsvpcConfiguration.assignPublicIp = 'ENABLED'
...
```

Reply:
The prior value is also returned in resources from policy.run()